### ISSUE TITLE
Suggestion: Reduce Logging Volumes

### ISSUE THREAD

**Issue by YPyltiai** [grzegorz914/homebridge-denon-tv#67]
Hi @grzegorz914,

Thanks for the plug-in!

It feels like there is a bit of redundant logging happening, which may be good to reduce or put into some controlled funnel & reduce logging level. Good example: https://github.com/grzegorz914/homebridge-denon-tv/blob/master/index.js#L375-L396 this information pops up every 5 (or whatever is set up as refresh interval) seconds in Homebridge Logs and completely takes over the whole log, so you cannot really see any important messages in the log anymore.

It would be ideal to make these as "debug" messages and add some variable into configuration to write or not debug logs. Alternatively just track errors and display those - "success" state of refresh seems to be a bit redundant for logging in this case.

Thanks!

**Comment by grzegorz914** [comment]
This is not true, this log info with device information is only show one time after homebridge restart.

**Comment by YPyltiai** [comment]
Is there any setting that may be influencing it? I made sure I do not run any debug logging on higher (homebridge) level. As a test - after commenting L375-L382 out I no longer see these entries in logs every 5 seconds.

It was a clean install of latest plugin version on latest stable HB.

**Comment by bigrichwood** [comment]
> This is not true, this log info with device information is only show one time after homebridge restart.

On mine it's shown every time you open the Home app on a device, and that's still the case with the latest release.

**Comment by grzegorz914** [comment]
Are U Both using docker image?

**Comment by bigrichwood** [comment]
No, standard install on a Pi.

**Comment by grzegorz914** [comment]
Please post the log here and config. As U see in the code this only show after reboot homebridge or after reconnect  the AV to the network.

**Comment by bigrichwood** [comment]
Log - Each of the timestamps correspond to me opening the Home app on my phone or laptop:

[14/02/2021, 13:35:58] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:35:58] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:35:58] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:36:52] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:36:52] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:36:52] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:38:23] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:38:23] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:38:23] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:38:46] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:38:46] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:38:46] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:38:47] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:38:47] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:38:47] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:40:24] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:40:24] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:40:24] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:40:34] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:40:34] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:40:34] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:40:35] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:40:35] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:40:35] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:40:41] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 13:40:41] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:40:41] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:41:36] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY
[14/02/2021, 13:41:36] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 13:41:36] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 14:55:53] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Volume level successful: -13 dB
[14/02/2021, 14:55:53] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Power state successful: ON
[14/02/2021, 14:55:53] [DenonTv] Device: 192.168.0.201 AV Receiver Main Zone, get current Input successful: AppleTV MPLAY

Config:
    "devices": [
                {
                    "name": "AV Receiver",
                    "host": "192.168.0.201",
                    "port": 80,
                    "refreshInterval": 5,
                    "zoneControl": 0,
                    "volumeControl": 2,
                    "masterPower": false,
                    "switchInfoMenu": false,
                    "inputs": [
                        {
                            "name": "TiVo",
                            "reference": "SAT/CBL",
                            "type": "HDMI",
                            "mode": "SI"
                        },
                        {
                            "name": "AppleTV",
                            "reference": "MPLAY",
                            "type": "HDMI",
                            "mode": "SI"
                        },
                        {
                            "name": "UHD",
                            "reference": "BD",
                            "type": "HDMI",
                            "mode": "SI"
                        },
                        {
                            "name": "Blu-Ray",
                            "reference": "DVD",
                            "type": "HDMI",
                            "mode": "SI"
                        },
                        {
                            "name": "FireTV",
                            "reference": "AUX2",
                            "type": "HDMI",
                            "mode": "SI"
                        }
                    ],
                    "modelName": "AVR-X3200"
                }
            ],
            "platform": "DenonTv"

**Comment by grzegorz914** [comment]
All is correct in your log.

**Comment by bigrichwood** [comment]
> All is correct in your log.

Other than it's logging everytime the status is queried, and not just when Homebridge is started?

In which case we're back to the original question, does it need to log so much? My preference would be for it to behave as all the other plugins I use, it logs at startup, and it logs any errors beyond that. Successful operations don't need a log entry in normal use.

**Comment by grzegorz914** [comment]
Please read the original case on first post, Your question hav nothing to do with the oryginal  problem. 
If U wish have more flexibility with logging I will add possibility to disable this logging in next release.

**Comment by YPyltiai** [comment]
@grzegorz914 in my case I have both logs I pointed to in code & what @bigrichwood sent. I assume the reason may be that in those lines that I mentioned you use just `log` and not something like `log.info` - maybe on Docker it has different behavior somehow.

For the time being I just patched it to be `log.debug` for my installation but as @bigrichwood  said - it would be great to have options to resuce logging.

Thanks!
